src/ark/analysis/cell_neighborhood_stats.py

```diff
@@ -0,0 +1,231 @@
+    all_diversity_data = reduce(
+        lambda left, right: pd.merge(left, right, on=[settings.FOV_ID, settings.CELL_LABEL]),
+        diversity_data)
+
+    return all_diversity_data
+
+
+def calculate_mean_distance_to_cell_type(
+        cell_table, dist_xr, cell_cluster, k, cell_type_col=settings.CELL_TYPE,
+        cell_label_col=settings.CELL_LABEL):
+    """Function to calculate mean distance of all cells to a specified cell type
+    Args:
+        cell_table (pd.DataFrame):
+            Dataframe containing all cells and their cell type
+        dist_xr (xr.array):
+            Cell by cell distances for all cells
+        cell_cluster (str):
+            Cell cluster to calculate distance to
+        k (int):
+            Number of nearest neighbours
+        cell_type_col (str):
+            column with the cell phenotype
+        cell_label_col (str):
+            column with the cell labels
+
+    Returns:
+        np.array:
+            mean distances for each cell to the cluster cells
+    """
+
+    # get cell ids for all cells of specific cluster
+    cluster_labels = cell_table.loc[cell_table[cell_type_col] == cell_cluster, cell_label_col]
+
+    # get all cells that match specified cell cluster
+    dist_xr = dist_xr.loc[:, dist_xr.dim_1.isin(cluster_labels)]
+
+    # keep the closest k values, not including itself
+    dist_xr = dist_xr.where(dist_xr > 0)
+    if dist_xr.shape[1] < k:
+        # image must contain at least k cell_cluster cells to receive an average dist
+        return [np.nan] * len(dist_xr)
+
+    sorted_dist = np.sort(dist_xr.values, axis=1)
+    sorted_dist = sorted_dist[:, :k]
+
+    # take the mean
+    mean_dists = sorted_dist.mean(axis=1)
+
+    return mean_dists
+
+
+def calculate_mean_distance_to_all_cell_types(
+        cell_table, dist_xr, k, cell_type_col=settings.CELL_TYPE,
+        cell_label_col=settings.CELL_LABEL):
+    """Wrapper function to calculate mean distance of all cells against all cell types
+    Args:
+        cell_table (pd.DataFrame):
+            Dataframe containing all cells and their cell type
+        dist_xr (xr.array):
+            Cell by cell distances for all cells
+        k (int):
+            Number of nearest neighbours
+        cell_type_col (str):
+            column with the cell phenotype
+        cell_label_col (str):
+            column with the cell labels
+
+    Returns:
+        pd.DataFrame:
+            average distances
+    """
+
+    # get all cell clusters in cell table
+    all_clusters = np.unique(cell_table[cell_type_col])
+
+    # call calculate_mean_distance_to_cell_type for all cell clusters
+    avg_dists = pd.DataFrame(index=cell_table.index.values, columns=all_clusters)
+    for cell_cluster in all_clusters:
+        avg_dists.loc[:, cell_cluster] = calculate_mean_distance_to_cell_type(
+            cell_table, dist_xr, cell_cluster, k, cell_type_col, cell_label_col)
+        
+    return avg_dists
+
+
+def cell_neighbor_distance_analysis(
```

Comment:
Even these top-level functions should have verb-based names. 

I think the `neighbor` is superfluous here, something like cell_distance_analysis describes it just as well and will help reduce confusion with the other functions